src/frontend/src/catalog/root_catalog.rs

```diff
@@ -45,6 +45,8 @@ pub struct Catalog {
     version: CatalogVersion,
     database_by_name: HashMap<String, DatabaseCatalog>,
     db_name_by_id: HashMap<DatabaseId, String>,
+    /// all table catalogs in the cluster identified by universal unique table id.
+    table_by_id: HashMap<TableId, TableCatalog>,
```

Comment:
Why not just add a field `fragment_id` in `TableDesc`, which is used in plan fragmenter? Introducing a duplicate table catalogs here cost too much here. 🤔

Reply:
We don't always have `TableDesc`. In `src/frontend/src/scheduler/distributed/stage.rs`, there is a function `chooseWorker`. We can only get a `PlanFragment` which just has a `associated_mview_id`. So we need a mapping from `TableId` to `Table` or `FragmentId`.

Reply:
Tianshuo and I think the duplicate catalogs here is acceptable. And one more thing, I think we should save every catalog id to catalog mapping in struct `Catalog` instead of `DatabaseCatalog`, `SchemaCatalog` and so on. Because our catalog id is unique.

Reply:
Okay, I understand. Let's keep it.